### ISSUE TITLE
pipe GET request to POST multipart 

### ISSUE THREAD

**Issue by mattador1** [request/request#1605]
Is it possible to make somethig like this.

```js
 var formData = {
       keyId: 'myId',
       attachment: request.get({uri : url, headers: {'Authorization': 'Basic ' + key}})
 };
 request.post({uri: 'post_url.com', formData: formData, headers: {'Authorization': 'Basic ' + key}
          
    }, function optionalCallback(err, httpResponse, body) {
})
```

Cause I want to fetch image and attach it to attachment, but I want to stream it as attachment. I don't want to load GET request (attachment) in memory on save it on hdd.
I'm trying this and server (which is third-party) response with bad request. When I do
attachment: fs.createReadStream(...path) all is ok and I can send attachment but than I have to save it to disk first. Is is possible to just pass it through. I want to have my server as transparent server(with possibility to add 'keyId' and than just forward it. 

Thx.

**Comment by simov** [comment]
It should be possible to do it. Did you tried to the `GET` request separately, to check if it returns the correct result? If you are expecting a binary file you should set `encoding:null` at least.

**Comment by mattador1** [comment]
When i tried this server returns 400 bad request. Also i did GET request separetly and it works I get image.

**Comment by simov** [comment]
Actually now I'm seeing the you are using basic auth, this might be related to a bug that was fixed previously, but not published on npm yet, can you try with the latest master instead?

**Comment by mattador1** [comment]
I tried with latest master 2.56.1, still no luck.
Again with attachment: fs.createReadStream(image) works, message uploads to server ok.
But when I use request.get(....   I get bad request response from server. Specifically I'm posting (email multipart) to mailgun and it seems something is wrong. Unfortunately I didn't get any useful response from mailgun support why I'm getting bad request.

**Comment by mattador1** [comment]
I've been thinking about this problem and maybe server where is POSTed requires content-length property of multipart/form-data and  because we are referencing attachment by GET request (stream) that info is not available at the time of POST. But maybe I'm wrong. I'm not really sure is request module first downloads whole attchment in memory through GET and than POST is executed or it streams attachment when POST-ing.

**Comment by simov** [comment]
Try this:

```js
request.post('https://api.mailgun.net/v2/[YOUR_DOMAIN]'+'/messages', {
  auth:{user:'api',pass:'[YOUR_API_KEY]'},
  formData:{
    from:'valid-email@gmail.com',
    to:'another-valid-email@gmail.com',
    subject:'Purest is awesome! (mailgun+attachments)',
    html:'<h1>Purest is awesome!</h1>',
    text:'True idd!',
    attachment:request.get('http://i.imgur.com/xEvqstE.png', {encoding:null})
  }
},
function (err, res, body) {
  debugger
  if (err) console.log(err)
  console.log(body)
})
```

Works for me.

**Comment by mattador1** [comment]
It works for me too, so there is obvious some error when fetching with GET request mailgun attachment, (cause imgur works) maybe because there is only path and not actual image in path for example path to mailgun attachment is something like this:
....a.mailgun.org/messages/WyI5MGQ5NDRmMWZiIiwgWyJkNzEyODlhYi0zZjk5LTQzODQtYjA2YS03M2FhNTU0OGRiYmQiXSwgIm1haWxndW4iLCAib2xkY29rZSJd/attachments/0
But then I tried this to form data:

var formData = {
            // Pass multiple values /w an Array
            from: 'valid@email.com',
            to: 'valid2@email.com',
            subject: 'SUBJECT',
            text: 'TEXT',
            attachment:  {
                
                value: request.get(url,{encoding: null, headers: {'Authorization': 'Basic ' + key}}),
                options: {
                    filename: 'test.png',
                    contentType: 'image/png'
                }
                
            }
But no luck still bad request.

**Comment by mattador1** [comment]
I did little research and I think maybe the problem is that request module can't POST attachment which is referenced by request.GET if the response from GET is chunked.
Here are response headers when fetching data (same image from different systems):

Imgur:
headers:  { 'last-modified': 'Sun, 31 May 2015 15:12:31 GMT',
  etag: '"f3d55d01a29de4550cb23989bb19dbd7"',
  'content-type': 'image/png',
  'cache-control': 'public, max-age=31536000',
  'content-length': '5538',
  'accept-ranges': 'bytes',
  date: 'Sun, 31 May 2015 15:46:56 GMT',
  age: '2062',
  connection: 'keep-alive',
  'x-served-by': 'cache-iad2137-IAD, cache-fra1237-FRA',
  'x-cache': 'HIT, MISS',
  'x-cache-hits': '1, 0',
  'x-timer': 'S1433087216.305591,VS0,VE87',
  'access-control-allow-methods': 'GET, OPTIONS',
  'access-control-allow-origin': '*',
  server: 'cat factory 1.0' }

Mailgun:
headers:  { server: 'nginx/1.7.9',
  date: 'Sun, 31 May 2015 15:51:40 GMT',
  'content-type': 'image/png',
  'transfer-encoding': 'chunked',
  connection: 'keep-alive',
  'content-disposition': 'attachment',
  'access-control-allow-origin': '*',
  'access-control-max-age': '600',
  'access-control-allow-methods': 'GET, POST, PUT, DELETE, OPTIONS',
  'access-control-allow-headers': 'Content-Type, x-requested-with' }

**Comment by mattador1** [comment]
Finally worked

I used this for formData: 
var formData = {
            from: 'valid@email.com',
            to: 'valid2@€mail.com,
            subject: 'Subject',
            text: 'text',
            attachment: {
                value:request.get(mail.url,{encoding: null, auth:{user:'api',pass:api_key}}),
                options: {
                    filename: 'test.png',
                    contentType: 'image/png',
                    knownLength: mail.size
                }
            }
and... for final POST
   request.post({uri: '.....mailgun.org/messages',formData: formData, auth:{user:'api',pass:api_key},
             headers:{'transfer-encoding': 'chunked'}}, function optionalCallback (err, httpResponse, body) {
             if (err) {....
Most important part here was specifying transfer-encoding part in POST and knowLength in GET request. Without this it won't work. Also question for someone with more http knowledge / author. Do I need to specify transfer-encoding header in above GET request (I tested and it works with and without it). 
Also one more for author, with above configuration does request module do just redirecting all traffic (mainly attachment cause they intend to get big) to POST address through stream. Or does app first GETs attachment (loads it whole in memory) and than POSTs it (with rest of data). I would like to have minimal impact on server memory and hdd.

Thx.
